### PR TITLE
Consolidate Pill-Shaped Button Variant

### PR DESCRIPTION
**Pull request by ayrock-dev**
#### Changeset
* Introduce `xs` as a small, pill-shaped button variant: `<Button size="xs"></Button>`
* Consistently use this new `<Button size="xs"></Button>` component for small, pill-shaped action buttons
* Constistently order action buttons when presented left-to-right, in the order of:
  * Like -> Share -> other actions
* Consistently use `<Button variant="secondary"></Button>` for action buttons
  * Previous usage was a variety of `"secondary"`, `"ghost"`, and `"outline"`
* Consistently use a "Calendar" icon before the relative date on: Solution Detail, Challenge Detail, Solution List Item, and Comments

## Related Issue

Implements #755

## Motivation and Context

Consistent UX creates a consistent look-and-feel, brand identity, and helps the user more quickly find actions as they become familiar with the expected order of actions button across the app.

## How Has This Been Tested?

Tested locally, in light and dark mode, on my local Chrome and Brave browsers. I'd appreciate a test in another browser if folks have time, ie Safari

## Screenshots

|**Before**|**After**|
|---|---|
|image|image|
|image|image|
|image|image|
|image|image|

## Video

Video better shows hover/active/selected states.

https://github.com/typehero/typehero/assets/7817695/c925868b-790b-417b-b038-8619aee7b2f0

### CODE REVIEW

**Comment by ayrock-dev** [comment]
I can't seem to assign reviewers or manage labels but I'm happy to label this one.

@PickleNik @bautistaaa looking for feedback, happy to address a feedback loop before moving this one in. i made some opinionated changes

**Comment by bautistaaa** [comment]
the only thing im not too keen on is 

CleanShot 2023-10-02 at 21 38 07@2x

those can all be on one row imo (share/upvote)

**Comment by ayrock-dev** [comment]
> the only thing im not too keen on is
> 
> those can all be on one row imo (share/upvote)

Totally, can-do. Tried to make that header on a Solution look the same as the header on a Challenge. I agree with you tho.

**Comment by ayrock-dev** [comment]
@bautistaaa @PickleNik Added video above to see it live in action. The video better shows hover/active/selected states.

File: packages/ui/src/components/button.tsx [hunk 1]
```diff
@@ -18,6 +18,7 @@ const buttonVariants = cva(
       },
       size: {
         default: 'h-10 py-2 px-4',
+        xs: 'h-6 px-2 rounded-full text-xs', // pill-shaped
```
Comment (ayrock-dev) [inline packages/ui/src/components/button.tsx:21]: this is the 🍖 and 🥔s of the change, the rest is basically usage and re-ordering elements

**Comment by bautistaaa** [comment]
can you fix the e2e tests <3

https://github.com/typehero/typehero/actions/runs/6388828715/job/17339288347?pr=760

**Comment by billc-dev** [comment]
The report doesn't get published if the e2e test fails.
In `quality.yml` for job `e2e`, can you add `if: always()` to step `Save PR number` and `Upload prNumber` as well

**Comment by ayrock-dev** [comment]
@billc-dev created #764 for this improvement

**Pull request merged**